### PR TITLE
Added accuracy validation tests, updated requirements.txt

### PR DESCRIPTION
**Pull request by pfinashx**
Added requirements.txt in tests/accuracy_validation directory
Added json file with model parameters
Added conftest.py
Added accuracy tests

### CODE REVIEW

**Comment by Ilya-Krylov** [comment]
Please rebase on develop, since significant changes were made. The `pytorch_toolkit/tests/` is for workaround only not and it is going to be removed as soon as CI is reconfigured. Please put your tests in `tests` dir.

**Comment by pfinashx** [comment]
@LeonidBeynenson @Ilya-Krylov Please take a look

File: ote/requirements.txt [hunk 1]
```diff
@@ -2,4 +2,3 @@ requests
 pyyaml
 mmcv
 texttable
-pytest
```
Comment (Ilya-Krylov) [inline ote/requirements.txt:5]: Could you do some useless change in pytorch_toolkit/tests/run_model_template_tests.sh like this https://github.com/openvinotoolkit/training_extensions/pull/460/commits/6dd4779b5c56f6282deda7a3956cee90231caad0
in order to run existing tests. CI has not been reconfigured properly yet. That's why you should do it.

I guess tests will be failed due to this change (removing `pytest`).

Comment (pfinashx) [reply ote/requirements.txt:5]: We have separate requirements.txt, and now all req's for accuracy validation stored in tests/accuracy_validation/requirements.txt including pytest.

Comment (Ilya-Krylov) [reply ote/requirements.txt:5]: I see, but pytest is needed here https://github.com/openvinotoolkit/training_extensions/blob/develop/tests/run_model_templates_tests.sh#L24

Comment (pfinashx) [reply ote/requirements.txt:5]: Ok, got it, will update soon!

**Comment by Ilya-Krylov** [comment]
@pfinashx @LeonidBeynenson I am ok with changes

@LeonidBeynenson I rely on you to approve these changes.

File: tests/accuracy_validation/test_accuracy_validation.py [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+                    if i is None:
+                        i = '-'
+                    with tag('td'):
+                        text(i)
+    f = open(PROJECT_ROOT / 'results.html', 'w')
+    f.write(doc.getvalue())
+    f.close()
+
+
+eval_config = json.load(open(TEST_ROOT / 'ote_accuracy_validation.json'), object_pairs_hook=OrderedDict)
```
Comment (LeonidBeynenson) [inline tests/accuracy_validation/test_accuracy_validation.py:164]: Pavel, may be it is better to make a pytest fixture for this block of code?
Probably with `scope="module"` and `autouse=True`

**Review by LeonidBeynenson — CHANGES_REQUESTED** [review]

File: tests/accuracy_validation/test_accuracy_validation.py (hunk 2 above)
Comment (LeonidBeynenson) [reply tests/accuracy_validation/test_accuracy_validation.py:164]: P.S. Or without `autouse` may be even better and more clean

File: tests/accuracy_validation/test_accuracy_validation.py [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+BG_COLOR_RED_HEX = 'ffcccc'
+
+DIFF_TARGET_MIN_GLOBAL = -0.1
+DIFF_TARGET_MAX_GLOBAL = 0.1
+
+TEST_ROOT = Path(__file__).parent
+PROJECT_ROOT = TEST_ROOT.parent.parent
+OPENVINO_DIR = PROJECT_ROOT.parent / 'intel' / 'openvino'
+if not os.path.exists(OPENVINO_DIR):
+    OPENVINO_DIR = PROJECT_ROOT.parent / 'intel' / 'openvino_2021'
```
[the remark this answers is not in the document]
Comment (druzhkov-paul) [reply tests/accuracy_validation/test_accuracy_validation.py:40]: @LeonidBeynenson please note that OpenVINO installation from PyPI is also possible, it requires a bit different logic to find it and derive paths to MO/AC/IE. Not sure if this option should also be handled by this code, but please keep it in mind.
Useful link: https://github.com/openvinotoolkit/open_model_zoo/pull/2190

[the remark this answers is not in the document]
Comment (pfinashx) [reply tests/accuracy_validation/test_accuracy_validation.py]: Done

File: tests/accuracy_validation/test_accuracy_validation.py [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+param_list = []
+ids_list = []
+model_dict = []
+row_dict = OrderedDict()
+color_dict = OrderedDict()
+
+
+@pytest.fixture(autouse=True, scope="module")
+def setup_ac():
+    subprocess.run("virtualenv -ppython3.8 acc_check", cwd=PROJECT_ROOT, check=True, shell=True)
```
[the remark this answers is not in the document]
Comment (pfinashx) [reply tests/accuracy_validation/test_accuracy_validation.py]: Done

File: tests/accuracy_validation/test_accuracy_validation.py [hunk 5]
```diff
@@ -0,0 +1,235 @@
...
+
+eval_config = json.load(open(TEST_ROOT / 'ote_accuracy_validation.json'), object_pairs_hook=OrderedDict)
+for sample_type in eval_config:
+    model_type = eval_config[sample_type]
+    for model_name in model_type:
+        model_dict = model_type[model_name]
+        for model in model_dict:
+            test_id = model_dict[model].get('test_id', {})
+            expected = model_dict[model].get('target', {})
+            alt_export = model_dict[model].get('subfolder') if not None else None
```
[the remark this answers is not in the document]
Comment (pfinashx) [reply tests/accuracy_validation/test_accuracy_validation.py]: Done

File: tests/accuracy_validation/test_accuracy_validation.py [hunk 6]
```diff
@@ -0,0 +1,235 @@
...
+    f.close()
+
+
+eval_config = json.load(open(TEST_ROOT / 'ote_accuracy_validation.json'), object_pairs_hook=OrderedDict)
+for sample_type in eval_config:
+    model_type = eval_config[sample_type]
+    for model_name in model_type:
+        model_dict = model_type[model_name]
+        for model in model_dict:
+            test_id = model_dict[model].get('test_id', {})
```
[the remark this answers is not in the document]
Comment (pfinashx) [reply tests/accuracy_validation/test_accuracy_validation.py]: Done

File: tests/accuracy_validation/test_accuracy_validation.py [hunk 7]
```diff
@@ -0,0 +1,235 @@
...
+
+def make_table_row(expected_, key, error_message, metric, diff_target):
+    if metric is not None:
+        row = [str(key), str(expected_), str(metric), str(diff_target), str("-")]
+    else:
+        row = [str(key), str(expected_), str("Not executed"), str("-"), str(error_message)]
+    return row
+
+
+def get_export_test_params(model, model_name, test_id, sample_type, type=None) -> Tuple[str, str, str, str]:
```
[the remark this answers is not in the document]
Comment (pfinashx) [reply tests/accuracy_validation/test_accuracy_validation.py]: Done

File: tests/accuracy_validation/test_accuracy_validation.py (hunk 2 above)
Comment (pfinashx) [reply tests/accuracy_validation/test_accuracy_validation.py:168]: It's does not works. All fixtures executed after collecting tests, and parameters will not be collected. We can use something about this:
`def get_test_params():
...
return param_list, ids_list


test_param_list, test_ids_list = get_test_params()

@pytest.mark.parametrize("model_, test_id_, sample_type_, model_name_, expected_, alt_export_", param_list,
                         ids=ids_list)
def test_eval(data_dir, model_, test_id_, sample_type_, model_name_, expected_, alt_export_):
...`

Comment (pfinashx) [reply tests/accuracy_validation/test_accuracy_validation.py]: Done

**Comment by LeonidBeynenson** [comment]
run tests

**Comment by LeonidBeynenson** [comment]
run tests

**Comment by pfinashx** [comment]
Rebased to develop

**Review by Ilya-Krylov — APPROVED** [review]

**Comment by Ilya-Krylov** [comment]
run tests

**Comment by pfinashx** [comment]
We have some issues in CI after weekend. As soon as all problems were resolved, I will run tests.

**Comment by Ilya-Krylov** [comment]
run tests

**Comment by Ilya-Krylov** [comment]
is it ready to merge?

**Comment by pfinashx** [comment]
> is it ready to merge?

Yes, ready!

**Pull request merged**